### ISSUE TITLE
Templates with a size greater than 51,200 bytes must be deployed via an S3 Bucket. 

### ISSUE THREAD

**Issue by maceto** [falnyr/aws-sam-deploy-action#4]
Hi Jan,

After made some additions to template.yml I got this message after try to deploy

Templates with a size greater than 51,200 bytes must be deployed via an S3 Bucket. Please add the --s3-bucket parameter to your command. The local template will be copied to that S3 bucket and then deployed.

My implementation

```
      - name: Deploy SAM template
        uses: falnyr/aws-sam-deploy-action@v1.2.1
        env:
          TEMPLATE: 'template.yaml'
          PARAMETER_OVERRIDES: DbUri=${{ secrets.DEV_DB_URI }} 
          NO_FAIL_EMPTY_CHANGESET: true
          AWS_STACK_NAME: dev-callaction-core
          AWS_REGION: ${{ secrets.AWS_REGION }}
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          AWS_DEPLOY_BUCKET: MyS3Bucket
```

is already implemented `AWS_DEPLOY_BUCKET: MyS3Bucket`.

Reading SAM documentation I found the definition of the param --s3-bucket

https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html

--s3-bucket: The URI of the Amazon S3 bucket where this command uploads your AWS CloudFormation template. This is required to deploy templates that are larger than 51,200 bytes.

Maybe AWS_DEPLOY_BUCKET is not referencing the option --s3-bucket on SAM Deploy command? 

Thank you in advance
Martin

**Comment by falnyr** [comment]
Hi @maceto, the `AWS_DEPLOY_BUCKET`  is indeed referencing the `--s3-bucket` option. https://github.com/falnyr/aws-sam-deploy-action/blob/6fa75d198b25e1f5a1f7b7dc51e167cac85b7978/entrypoint.sh#L84

Can you share any specific error message so that we can resolve the issue?

Thanks

**Comment by maceto** [comment]
Hi @falnyr, I was trying to prepare the PR with the fix.

The problem is with `aws cloudformation deploy` command, no `aws cloudformation package`. The output of `aws cloudformation package` creates a template larger than 51,200 bytes. So `aws cloudformation deploy` needs a S3 bucket to upload the template before deploy, is the option --s3-bucket for `aws cloudformation deploy` . Make sense?

Here the doc for deploy 

https://docs.aws.amazon.com/cli/latest/reference/cloudformation/deploy/index.html#deploy

Thank you
Martin

**Comment by maceto** [comment]
Hi @falnyr,

Here the PR https://github.com/falnyr/aws-sam-deploy-action/pull/5 with the change. Let me know what do you think about the change.

Thank you
Martin

**Comment by falnyr** [comment]
I'm just reading that I overlooked the `--s3-bucket` option for `deploy` instead of `package`. Sorry about that. Added a comment to your PR.

**Comment by falnyr** [comment]
solved by PR #5.
